pkg/operator/controller/credentials-secret/controller_test.go

```diff
@@ -131,6 +135,106 @@ func TestReconcile(t *testing.T) {
 			inputRequest:    testRequest(),
 			expectedResult:  reconcile.Result{},
 		},
+		{
+			name:            "Target secret has expected keys for Infoblox provider",
+			existingObjects: []runtime.Object{testInfobloxExtDNSInstance(), testInfoBloxSrcSecret(), testInfoBloxTargetSecret()},
+			inputConfig:     testConfig(),
+			inputRequest:    testRequest(),
+			expectedResult:  reconcile.Result{},
+			expectedEvents: []test.Event{
+				{
+					EventType: watch.Added,
+					ObjType:   "secret",
+					NamespacedName: types.NamespacedName{
+						Namespace: testOperandNamespace,
+						Name:      testTargetSecretName,
+					},
+				},
+			},
+		},
+		{
+			name:            "Target secret doesn't have expected keys for Infoblox provider",
+			existingObjects: []runtime.Object{testInfobloxExtDNSInstance(), testInfobloxWrongSrcSecret(), testInfoBloxTargetSecret()},
+			inputConfig:     testConfig(),
+			inputRequest:    testRequest(),
+			expectedResult:  reconcile.Result{},
+			errExpected:     true,
+		},
+		{
+			name:            "Target secret has expected keys for Azure provider",
+			existingObjects: []runtime.Object{testAzureExtDNSInstance(), testAzureSrcSecret(), testAzureTargetSecret()},
+			inputConfig:     testConfig(),
+			inputRequest:    testRequest(),
+			expectedResult:  reconcile.Result{},
+			expectedEvents: []test.Event{
+				{
+					EventType: watch.Added,
```

Comment:
Same here, the test case can be `Modified` at max (if the value of the key is different), no event at best.